lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart

```diff
@@ -64,6 +64,15 @@ final class ViewFocusBinding {
   });
 
   late final DomEventListener _handleFocusout = createDomEventListener((DomEvent event) {
+    // During focusout processing, activeElement typically points to <body />.
+    // However, if an element is focused during a blur event, activeElement points to that focused element.
+    // We leverage this behavior to ignore focusout events where the document has focus but activeElement is not <body />.
+    //
+    // Refer to https://github.com/flutter/engine/pull/54965 for more info.
+    if (domDocument.hasFocus() && domDocument.activeElement != domDocument.body) {
```

Comment:
For long conditionals like this, I find it more readable to put the result in a variable with a meaningful name. E.g:
```suggestion
    final bool wasFocusInvoked = domDocument.hasFocus() && domDocument.activeElement != domDocument.body;
    if (wasFocusInvoked) {
```